src/main/scala/edu/ie3/osmogrid/lv/region_coordinator/OsmoGridModelPartitioner.scala

```diff
@@ -0,0 +1,180 @@
+/*
+ * © 2022. TU Dortmund University,
+ * Institute of Energy Systems, Energy Efficiency and Energy Economics,
+ * Research group Distribution grid planning and operation
+ */
+
+package edu.ie3.osmogrid.lv.region_coordinator
+
+import edu.ie3.osmogrid.model.OsmoGridModel.{EnhancedOsmEntity, LvOsmoGridModel}
+import edu.ie3.util.geo.GeoUtils
+import edu.ie3.util.osm.model.OsmEntity
+import edu.ie3.util.osm.model.OsmEntity.Relation.RelationMemberType
+import org.locationtech.jts.geom.Polygon
+import edu.ie3.osmogrid.lv.region_coordinator.EntityAllocationStrategy
+
+import scala.collection.parallel.{ParMap, ParSeq}
+
+object OsmoGridModelPartitioner {
+
+  /** Id of the boundary relation
+    */
+  type AreaKey = Long
+
+  /** Assign all entities of given OsmoGridModel to the given areas. Depending
+    * on the type of entity they are assigned to more than one area.
+    *
+    * @param osmoGridModel
+    *   the OsmoGridModel to partition
+    * @param areas
+    *   the areas that the entities should be assigned to
+    * @return
+    *   a map from area id (id of the boundary relation) to OsmoGridModel
+    */
+  def partition(
+      osmoGridModel: LvOsmoGridModel,
+      areas: ParMap[AreaKey, Polygon]
+  ): ParMap[AreaKey, LvOsmoGridModel] = {
+    val buildings = assign(
+      osmoGridModel.buildings,
+      areas,
+      EntityAllocationStrategy.AssignByMax
+    )
+    val highways = assign(
+      osmoGridModel.highways,
+      areas,
+      EntityAllocationStrategy.AssignToAll
+    )
+    val landuses = assign(
+      osmoGridModel.landuses,
+      areas,
+      EntityAllocationStrategy.AssignToAll
+    )
+    val boundaries = assign(
+      osmoGridModel.boundaries,
+      areas,
+      EntityAllocationStrategy.AssignByMax
```

Comment:
might it be sensible to use `AssignToAll` here as well?

Reply:
Well as discussed, boundaries of each level should be partitions of the level above. So there should be exactly one fit, except for the points of the boundary polygon which are shared with the boundary above (which is also part of the neighboring boundary, depending on the polygon matching method used).